### DISCUSSION TITLE
Tool Calling - how can we request to the LLM that it re-calls for data after a period of time

### DISCUSSION THREAD

**Discussion by algotechnology** [langchain4j/langchain4j#1604] [Q&A]
In my use case the data changes and the LLM keeps returning and using cached data ....

any ideas or tips to avoid this ?

Thanks

**Comment by InAnYan** [comment]
I've also been thinking about those kinds of problems.

You need to understand, that an LLM is just a `Text -> Text` function. And all other logic ("architecture" of an LLM app) should be managed externally.

If you have a tool call, that LLM uses, then it's the responsibility of the programmer who made the system to handle the tool call.

Not LLM returns output, but a combination of LLM + tools + servers that handle requests.

If your system returns cached output, then look into your caching mechanism. Turn if off, or make the expiry less

**Comment by algotechnology** [comment]
Thanks.  My tool side of things always returns the latest / live data.  The issue is that if the LLM has called it once it often caches the tool result in the LLM so the LLM does not call the tool for every request (which I would like).  Ive tried adding plenty of hints via @Tool which helps a bit but still it caches some data / fails to re-execute the actual tool call (if that makes sense).

**Reply by algotechnology** [reply]
Yes.  I tried to give it hints there and in the @Tool description.  I understand the issues but it might be a useful feature to add a 'valid until' date or something similar to the tool replies as at some point perhaps the LLM providers will support this as the LLMs continue to get used in a wider variety of applications.  I will continue to add phrases to System and Tool and report back if any particular phrasing helps!

**Reply by algotechnology** [reply]
I will have to try to go to that level.  Just been using the services (which work a treat).  Its just the LLM answer is caching data and then effectively displaying it when in reality it is stale and the tool would provide the correct data.  Interesting idea about TTL tho - that might be a good idea anyway to be honest.  In fast moving data situations that may well be an excellent approach.  In my case a TTL of 15 seconds would be ideal!
